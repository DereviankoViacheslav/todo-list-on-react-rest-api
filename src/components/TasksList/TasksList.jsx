import React from 'react';
import Task from '../Task';
import PropTypes from 'prop-types';
import './TasksList.scss';

function TasksList({ tasks, onDelete, onToggleDone }) {
  const tasksList = [...tasks]
    .sort((a, b) => a.done - b.done)
    .map((task) => {
      return (
        <Task key={task.id} {...task} onDelete={onDelete} onToggleDone={onToggleDone} />
      );
    });

  return (
    <ul className="list">
      {tasksList}
    </ul>
  );
}

TasksList.propTypes = {
  tasks: PropTypes.array.isRequired,
  onDelete: PropTypes.func.isRequired,
  onToggleDone: PropTypes.func.isRequired,
}

export default TasksList;