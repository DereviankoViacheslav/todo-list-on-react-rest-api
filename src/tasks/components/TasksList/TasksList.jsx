import './TasksList.scss';
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import * as tasksActions from '../../tasks.actions';
import { sortedTasksListSelector } from '../../tasks.selectors';
import Task from '../Task';
import PropTypes from 'prop-types';

const TasksList = ({ tasks, deleteTask, updateTask, getTasksList }) => {
  useEffect(() => {
    getTasksList();
  }, []);

  return (
    <ul className="list">
      {tasks.map((task) => {
        return (
          <Task key={task.id} {...task} onDelete={deleteTask} onToggleDone={updateTask} />
        );
      })}
    </ul>
  );
}

TasksList.propTypes = {
  tasks: PropTypes.arrayOf(PropTypes.shape()),
  getTasksList: PropTypes.func.isRequired,
  updateTask: PropTypes.func.isRequired,
  deleteTask: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  return {
    tasks: sortedTasksListSelector(state),
  };
};

const mapDispatchToProps = {
  getTasksList: tasksActions.getTasksList,
  updateTask: tasksActions.updateTask,
  deleteTask: tasksActions.deleteTask,
};

export default connect(mapStateToProps, mapDispatchToProps)(TasksList);