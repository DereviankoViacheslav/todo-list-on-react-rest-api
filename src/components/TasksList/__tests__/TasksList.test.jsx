import React from 'react';
import { shallow } from 'enzyme';
import TasksList from '../TasksList';

describe('<TasksList />', () => {
  it('should display tasks list sorted by "done"', () => {
    const props = {
      tasks: [
        { id: 'id-0', text: 'Task 1', done: true },
        { id: 'id-1', text: 'Task 2', done: false },
        { id: 'id-2', text: 'Task 3', done: true },
      ],
      onDelete: jest.fn(),
      onToggleDone: jest.fn(),
    }
    const wrappedComponent = shallow(<TasksList {...props} />);
    expect(wrappedComponent).toMatchSnapshot();
  });
});