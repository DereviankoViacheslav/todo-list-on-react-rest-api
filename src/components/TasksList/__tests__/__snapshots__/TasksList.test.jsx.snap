// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksList /> should display tasks list sorted by "done" 1`] = `
<ul
  className="list"
>
  <Task
    done={false}
    id="id-1"
    key="id-1"
    onDelete={[MockFunction]}
    onToggleDone={[MockFunction]}
    text="Task 2"
  />
  <Task
    done={true}
    id="id-0"
    key="id-0"
    onDelete={[MockFunction]}
    onToggleDone={[MockFunction]}
    text="Task 1"
  />
  <Task
    done={true}
    id="id-2"
    key="id-2"
    onDelete={[MockFunction]}
    onToggleDone={[MockFunction]}
    text="Task 3"
  />
</ul>
`;
